src/toxcore/dht/kbucket.rs

```diff
@@ -262,8 +262,8 @@ impl Bucket {
     }
 
     /// Check if node with given PK is in the `Bucket`.
-    pub fn contains(&self, pk: &PublicKey) -> bool {
-        self.nodes.iter().any(|n| &n.pk == pk)
+    pub fn contains(&self, base_pk: &PublicKey, pk: &PublicKey) -> bool {
+        self.nodes.binary_search_by(|n| base_pk.distance(&n.pk, pk)).is_ok()
```

Comment:
Why would you do that? If you want to reduce N to log N then ... nobody cares because N = 8

Reply:
Why not? We use binary search everywhere but not here, why? I like consistency. Besides bucket not necessary has size of 8. Th biggest bucket we have at the moment - 32.

Reply:
For instance this method uses binary search and takes base_pk: https://github.com/tox-rs/tox/blob/776a07f6e4d466fec562b1f1c879452baf4684b7/src/toxcore/dht/kbucket.rs#L156-L158
So when I stumbled when tried to use `contains` method with 2 keys in tests - suddenly it took only one key. So for consistency I decided to change it.

Reply:
OK, let it be